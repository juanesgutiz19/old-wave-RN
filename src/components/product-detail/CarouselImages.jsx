import React from 'react'
import { View, ActivityIndicator ,StyleSheet} from 'react-native'
import {Image} from 'react-native-elements'
import { Pagination } from 'react-native-snap-carousel'
import Carousel from 'react-native-snap-carousel'
import { size} from 'lodash'

export const CarouselImages = ({images, height, width, activeSlide, setActiveSlide}) => {
    const renderItem = ({item}) => {
        return (       
            <Image
                style={{width, height}}
                PlaceholderContent={<ActivityIndicator color='#fff'/>}
                source={{uri:'https://http2.mlstatic.com/D_684803-MLA44155935502_112020-O.jpg'}}
                resizeMode='contain'
            />
        )
    }
    return (
        <View>
            <Carousel
                layout={"default"}
                data={images}
                sliderWidth={width}
                itemWidth={width}
                itemHeight={height}
                renderItem={renderItem}
                onSnapToItem={(index)=> setActiveSlide(index)}
                style={styles.imageContainer}
            />
            <MyPagination data={images} activeSlide={activeSlide}/>
        </View>
    )
}

function MyPagination({data, activeSlide}){
    return (
        <Pagination dotsLength={size(data)}
        activeDotIndex={activeSlide}
        containerStyle={styles.containerPagination}
        dotStyle={styles.dotActive}
        inactiveDotStyle={styles.dotInactive}
        inactiveDotOpacity={0.6}
        inactiveDotScale={0.4}/>
    )
}

const styles = StyleSheet.create({
    containerPagination:{
        backgroundColor: 'transparent',
        zIndex: 1,
        position: 'absolute',
        bottom:0,
        alignSelf: 'center'
    },
    dotActive:{
        width: 15,
        height: 15,
        borderRadius: 10,
        marginHorizontal:2,
        backgroundColor: '#7444FB'
    },
    dotInactive:{
        width: 14,
        height: 14,
        borderRadius: 7,
        marginHorizontal: 2,
        backgroundColor: '#fff'
    },
    imageContainer: {
        flex: 1,
        alignSelf: 'center',
        width: '100%',
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 10,
        },
        shadowOpacity: 0.24,
        elevation: 9,
    }
})