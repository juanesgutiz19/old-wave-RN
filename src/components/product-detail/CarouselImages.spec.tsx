
import React from "react"
import renderer from "react-test-renderer"

import { CarouselImages } from "./CarouselImages";

jest.runAllTimers();

describe("<CarouselImages />", () => {

    const tree: any = renderer.create(<CarouselImages
        images={[]}
        height={100}
        width={100}
        activeSlide={2}
        setActiveSlide={() => { }}
    />).toJSON()

    it("verify has 1 child", async (done) => {
        expect(tree?.children?.length).toBe(1);
        done();
    });

    it("verify type of <CarouselImages /> ", (done) => {
        expect(tree?.type).toBe("View");
        done();
    });

    it("verify size of item in <CarouselImages /> children", (done) => {

        const expectedSize = { itemWidth: 100, itemHeight: 100 };

        expect(tree?.children[0].props).toMatchObject(expectedSize);
        done();
    });

    it("verify type of <CarouselImages /> children", (done) => {
        expect(tree?.children[0].type).toBe("RCTScrollView");
        console.log(tree.children)
        done();
    });

    it('snapshot', () => {
        expect(tree).toMatchSnapshot();
    });


})