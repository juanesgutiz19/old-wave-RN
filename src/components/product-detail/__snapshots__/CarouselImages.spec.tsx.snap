// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselImages /> snapshot 1`] = `
<View>
  <RCTScrollView
    activeAnimationOptions={null}
    activeAnimationType="timing"
    activeSlideAlignment="center"
    activeSlideOffset={20}
    apparitionDelay={0}
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDelay={1000}
    autoplayInterval={3000}
    callbackOffsetMargin={5}
    containerCustomStyle={Object {}}
    contentContainerCustomStyle={Object {}}
    contentContainerStyle={
      Array [
        Object {
          "paddingLeft": 0,
          "paddingRight": 0,
        },
        Object {},
      ]
    }
    data={Array []}
    decelerationRate="fast"
    directionalLockEnabled={true}
    disableVirtualization={false}
    enableMomentum={false}
    enableSnap={true}
    firstItem={0}
    forwardedRef={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    hasParallaxImages={false}
    horizontal={true}
    inactiveSlideOpacity={0.7}
    inactiveSlideScale={0.9}
    inactiveSlideShift={0}
    initialNumToRender={6}
    invertStickyHeaders={false}
    inverted={false}
    itemHeight={100}
    itemWidth={100}
    keyExtractor={[Function]}
    layout="default"
    lockScrollTimeoutDuration={1000}
    lockScrollWhileSnapping={false}
    loop={false}
    loopClonesPerSide={3}
    maxToRenderPerBatch={13}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onResponderRelease={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onSnapToItem={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    overScrollMode="never"
    pinchGestureEnabled={false}
    removeClippedSubviews={true}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollsToTop={false}
    shouldOptimizeUpdates={true}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    slideStyle={Object {}}
    sliderWidth={100}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexDirection": "row",
        "opacity": 0,
        "width": 100,
      }
    }
    swipeThreshold={20}
    updateCellsBatchingPeriod={50}
    useScrollView={false}
    vertical={false}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={13}
  >
    <View />
  </RCTScrollView>
</View>
`;
